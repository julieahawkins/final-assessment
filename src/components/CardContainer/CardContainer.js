import React from 'react';
import { connect } from 'react-redux';
import Card from '../Card/Card';
import PropTypes, { array } from 'prop-types';

const CardContainer = ({ houseData }) => {
  const cards = houseData.map((house, index) => 
    <Card 
      key={`card-${index}`} 
      houseData={houseData[index]} />
  );

  return (
    <div className='Container'>
      {cards}
    </div>
  );
};

CardContainer.propTypes = {
  houseData: array.isRequired
};

const mapStateToProps = ({ houseData }) => ({ houseData });

export default connect(mapStateToProps, null)(CardContainer);