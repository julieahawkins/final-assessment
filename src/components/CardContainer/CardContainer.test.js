import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';
import configureMockStore from 'redux-mock-store';

describe('CardContainer test', () => {
  const fakeStore = configureMockStore()({
    data: []
  });

  const setup = () => {
    const wrapper = shallow(
      <Provider store={fakeStore}>
        <CardContainer />
      </Provider>
    );
    const Component = wrapper.find(CardContainer);

    return {
      Component
    };
  };

  it('should match its snap shot', () => {

    expect(setup.wrapper).toMatchSnapshot();
  });
})