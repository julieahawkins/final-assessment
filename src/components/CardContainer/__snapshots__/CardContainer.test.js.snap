// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer test should match its snap shot 1`] = `undefined`;
